internal/batches/ui/task_exec_tui.go

```diff
@@ -427,50 +428,102 @@ func diffStatDiagram(stat diff.Stat) string {
 }
 
 type stepsExecTUI struct {
+	out             *output.Output
 	task            *executor.Task
 	updateStatusBar func(string)
 }
 
 func (ui stepsExecTUI) ArchiveDownloadStarted() {
 	ui.updateStatusBar("Downloading archive")
+	ui.out.Verbosef("[%s] Downloading repository archive...", ui.task.Repository.Name)
 }
-func (ui stepsExecTUI) ArchiveDownloadFinished(err error) {}
+
+func (ui stepsExecTUI) ArchiveDownloadFinished(err error) {
+	if err != nil {
+		ui.out.Verbosef("[%s] Archive download failed: %v", ui.task.Repository.Name, err)
+	} else {
+		ui.out.Verbosef("[%s] Archive download completed", ui.task.Repository.Name)
+	}
+}
+
 func (ui stepsExecTUI) WorkspaceInitializationStarted() {
 	ui.updateStatusBar("Initializing workspace")
+	ui.out.Verbosef("[%s] Initializing workspace...", ui.task.Repository.Name)
+}
+
+func (ui stepsExecTUI) WorkspaceInitializationFinished() {
+	ui.out.Verbosef("[%s] Workspace initialization completed", ui.task.Repository.Name)
 }
-func (ui stepsExecTUI) WorkspaceInitializationFinished() {}
+
 func (ui stepsExecTUI) SkippingStepsUpto(startStep int) {
 	switch startStep {
 	case 1:
 		ui.updateStatusBar("Skipping step 1. Found cached result.")
+		ui.out.Verbosef("[%s] Skipping step 1 (cached result found)", ui.task.Repository.Name)
```

Comment:
Won't we have duplicate output when using `-v`?

```
[              Skipping step 1. Found cached result.             ]
[sourcegraph/sourcegraph] Skipping step 1 (cached result found)
```

or can you provide a screenshot of how this looks? with and without the verbose flag